### PR TITLE
added usage example for permission boundary.

### PR DESCRIPTION
**Pull request by ykethan**
#### Description of changes:
Added an example for permissions boundary using aws managed policy for DynamoDb

#### Related GitHub issue #, if available:
closes: https://github.com/aws-amplify/docs/issues/4042

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [x] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br />
      _ref: MDX: `[link](https://link.com)`
            HTML: `<a href="https://link.com">link</a>`_
            
### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/cli/project/permissions-boundary.mdx [hunk 1]
```diff
@@ -48,4 +53,34 @@ The IAM Permissions Boundary will be applied on the next `amplify push`.
...
+
+For this example lets consider a scenario where the team wants to create a environment thats allows the developers to only work with DynamoDB resources. We can use the [AWS managed](https://us-east-1.console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess$serviceLevelSummary?section=permissions) policy for DynamoDB that allows users to add a DynamoDB table.
+
+Add Permissions Boundary for the new environment, run:
+
+```
+amplify env add --permissions-boundary arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess
+```
+
+We can verify if the permissions boundary have been applied by checking the AWS IAM console for the `authRole` and `unauthRole` created by Amplify CLI.
```
Comment (edwardfoyle) [inline src/pages/cli/project/permissions-boundary.mdx:70]:
```suggestion
Verify the permissions boundary have been applied by checking the AWS IAM console for the `authRole` and `unauthRole` created by Amplify CLI.
```

**Review by edwardfoyle — CHANGES_REQUESTED** [review]
Left a few comments but my main feedback is that permission boundaries are not meant to scope down what a developer can do in a project but rather what a role that the CLI creates can do. To scope down what a developer can do, the permissions attached to the IAM user/role that the developer is using should be adjusted accordingly. The permission boundary is a safeguard against, say the CLI creating a lambda execution role in a dev environment that accidentally grants access to a prod DynamoDB table.

File: src/pages/cli/project/permissions-boundary.mdx [hunk 2]
```diff
@@ -48,4 +53,34 @@ The IAM Permissions Boundary will be applied on the next `amplify push`.
...
\ No newline at end of file
+Amplify CLI cannot automatically apply the existing boundary to a new environment in a different AWS account if the `--yes` flag is used during `amplify env add`. In this case, a new Permissions Boundary must be specified using `amplify env add --yes --permissions-boundary <IAM Policy ARN>` or to explicitly remove the Permissions Boundary from the new environment use `amplify env add --yes --permissions-boundary ''`.
+
+## Permissions Boundary usage example
+
+When working with different environments you may want to create an environment that allow users to only access and make modifications to specific resources.
+
+For this example lets consider a scenario where the team wants to create a environment thats allows the developers to only work with DynamoDB resources. We can use the [AWS managed](https://us-east-1.console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess$serviceLevelSummary?section=permissions) policy for DynamoDB that allows users to add a DynamoDB table.
```
[the remark this answers is not in the document]
Comment (ykethan) [reply src/pages/cli/project/permissions-boundary.mdx]: Thank you for the info @edwardfoyle, modified the use case accordingly.

File: src/pages/cli/project/permissions-boundary.mdx [hunk 3]
```diff
@@ -48,4 +53,32 @@ The IAM Permissions Boundary will be applied on the next `amplify push`.
...
+
+Suppose you want to scope down permissions for Amplify CLI to only allow an environment to work with DynamoDB tables. In this case, the [DynamoDB FullAccess Managed Policy](https://us-east-1.console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess$serviceLevelSummary?section=permissions) can be used as a permission boundary.
+
+To add a Permissions Boundary for a new environment, run:
+
+```
+amplify env add --permissions-boundary arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess
+```
+
+Verify if the permissions boundary have been applied by checking the AWS IAM console for the `authRole` and `unauthRole` created by Amplify CLI.
```
Comment (sdstolworthy) [inline src/pages/cli/project/permissions-boundary.mdx:68]:
```suggestion
Verify if the permissions boundary has been applied by checking the AWS IAM console for the `authRole` and `unauthRole` created by Amplify CLI.
```

File: src/pages/cli/project/permissions-boundary.mdx [hunk 4]
```diff
@@ -48,4 +53,32 @@ The IAM Permissions Boundary will be applied on the next `amplify push`.
...
+
+Add a DynamoDB table, run:
```
Comment (sdstolworthy) [inline src/pages/cli/project/permissions-boundary.mdx:70]:
```suggestion
To add a DynamoDB table, run:
```

File: src/pages/cli/project/permissions-boundary.mdx [hunk 5]
```diff
@@ -48,4 +53,32 @@ The IAM Permissions Boundary will be applied on the next `amplify push`.
...
+
+```
+amplify add storage
+```
+
+Select `NoSQL Database` and finish the setup wizard.
+
+Push the resources, run:
```
Comment (sdstolworthy) [inline src/pages/cli/project/permissions-boundary.mdx:78]:
```suggestion
To push the resources, run:
```

File: src/pages/cli/project/permissions-boundary.mdx [hunk 6]
```diff
@@ -35,11 +38,13 @@ If Amplify CLI is not able to automatically apply the Permissions Boundary to th
 ## Update an environment's Permissions Boundary
 
 To modify the Permissions Boundary of the current environment, run:
+
 ```
 amplify env update
 ```
 
-In non-interactive shells use 
+In non-interactive shells use
```
Comment (sdstolworthy) [inline src/pages/cli/project/permissions-boundary.mdx:46]:
```suggestion
In non-interactive shells, use:
```

File: src/pages/cli/project/permissions-boundary.mdx [hunk 7]
```diff
@@ -48,4 +53,32 @@ The IAM Permissions Boundary will be applied on the next `amplify push`.
...
+
+Select `NoSQL Database` and finish the setup wizard.
+
+Push the resources, run:
+
+```
+amplify push
+```
+
+If we create resources that are outside the scope of allowed resources on permissions boundary, for example a Cognito resource. The next `amplify push` will fail.
```
Comment (sdstolworthy) [inline src/pages/cli/project/permissions-boundary.mdx:84]: Is Permissions Boundary a proper noun or not? The casing is mixed in the documentation. In these new additions, it is lower cased "permissions boundary". In the existing documentation, it is Title Case in the existing language in this file. 

https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html

The docs here lead me to believe it should be lower case in all uses.

**Review by sdstolworthy — CHANGES_REQUESTED** [review]
Mostly nits, but the mixed casing should be addressed and made uniform.

**Review by josefaidt — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by sdstolworthy — APPROVED** [review]

**Review by edwardfoyle — CHANGES_REQUESTED** [review]
Previous comment is still relevant

**Comment by ykethan** [comment]
closing this to create a new PR.

**Pull request closed without merging**